platform/config/ensureDeepObject.ts

```diff
@@ -1,29 +1,42 @@
 const separator = '.';
 
+/**
+ * Recursively traverses through the object's properties and expands ones with
+ * dot-separated names into nested objects (eg. { a.b: 'c'} -> { a: { b: 'c' }).
+ * @param {*} obj Object to traverse through
+ * @returns {Object} Same object instance with expanded properties.
```

Comment:
Yeah, I'm not sure too. In this case it's actually even useless. It's just a habit from the past when I used Google Closure compiler to generate docs (for public API) :) But with Typescript I believe we'll have better tools that will be able to detect all types (hopefully, I'll need to check what is out there).